Library/ViewModels/PledgeViewModelTests.swift

```diff
@@ -1,179 +1,444 @@
 import Foundation
+@testable import KsApi
+@testable import Library
 import Prelude
 import ReactiveExtensions
+import ReactiveExtensions_TestHelpers
 import ReactiveSwift
 import XCTest
 
-@testable import KsApi
-@testable import Library
-import ReactiveExtensions_TestHelpers
-
+// swiftlint:disable line_length
 final class PledgeViewModelTests: TestCase {
-  private let vm: PledgeViewModelType = PledgeViewModel()
+  private var vm: PledgeViewModelType!
 
-  private let configureSummaryCellWithProject = TestObserver<Project, Never>()
-  private let configureSummaryCellWithPledgeTotal = TestObserver<Double, Never>()
-  private let project = TestObserver<Project, Never>()
-  private let reward = TestObserver<Reward, Never>()
-  private let isLoggedIn = TestObserver<Bool, Never>()
-  private let isShippingEnabled = TestObserver<Bool, Never>()
-  private let total = TestObserver<Double, Never>()
+  private let configureShippingLocationCellWithDataIsShippingRulesLoading = TestObserver<Bool, Never>()
+  private let configureShippingLocationCellWithDataProject = TestObserver<Project, Never>()
+  private let configureShippingLocationCellWithDataSelectedShippingRule = TestObserver<ShippingRule?, Never>()
 
-  private let reload = TestObserver<Bool, Never>()
+  private let configureSummaryCellWithDataPledgeTotal = TestObserver<Double, Never>()
+  private let configureSummaryCellWithDataProject = TestObserver<Project, Never>()
+
+  private let pledgeViewDataAndReloadIsLoggedIn = TestObserver<Bool, Never>()
+  private let pledgeViewDataAndReloadIsShippingEnabled = TestObserver<Bool, Never>()
+  private let pledgeViewDataAndReloadIsShippingRulesLoading = TestObserver<Bool, Never>()
+  private let pledgeViewDataAndReloadProject = TestObserver<Project, Never>()
+  private let pledgeViewDataAndReloadReload = TestObserver<Bool, Never>()
```

Comment:
Can I suggest adding a comment here for future Justin and Pavel, something like:

```
/*
`pledgeViewDataAndReload` emits the initial data for this view model along with a `Bool` indicating whether the table view should be reloaded (currently only the first time but probably during login as well). Every time any of the combined data of `PledgeViewData` changes, this signal will emit all of its data in order to ensure that the data source has the current state of the data that the cells have. This ensures that if a cell is recycled it will be reloaded with its most recent data. Given the noise that this frequent emission creates and also that what we really care about is the final emission from this signal, we only test its last value.
*/
```

Maybe this should actually be a comment above the VM output?

Reply:
I'll put it on the VM above the output